activerecord/lib/active_record/relation.rb

```diff
@@ -22,10 +22,11 @@ class Relation
     alias :loaded? :loaded
     alias :locked? :lock_value
 
-    def initialize(klass, table, predicate_builder, values = {})
+    def initialize(klass, table, predicate_builder, values = {}, connection_name = nil)
```

Comment:
Why are we injecting the connection name instead of the connection value directly? We should be able to retrieve the connection at the point where the relation is created, shouldn't we?